### ISSUE TITLE
checkout dies with "Error: Object to checkout does not match repository"

### ISSUE THREAD

**Issue by abliss** [twosigma/git-meta#204]
```
$ git meta checkout upstream/master
Error: Object to checkout does not match repository
    at Error (native)
```

This is not a very helpful message.

**Comment by bpeabody** [comment]
Yeah, that's from `libgit2`. How do you make it happen?

**Comment by abliss** [comment]
```
git meta version
version 0.4.0
internal build 5
```

`git status` shows that my meta repo is clean, and already detached at the HEAD of upstream/master.

**Comment by abliss** [comment]
Actually this may already be fixed by #199

**Comment by bpeabody** [comment]
Yeah, that's exactly what it is. My vacation-addled brain didn't connect the dots.
